### ISSUE TITLE
Having multiple tabs of different versions KIE Sandbox open may cause issues with the Shared Worker

### ISSUE THREAD

**Issue by thiagoelg** [apache/incubator-kie-issues#82]
Since a lot of our processing in the frontend is done via Shared Worker most releases will need to update it.
That can be an issue when the user has multiple tabs of the KIE Sandbox open and some of them are on older versions, making the Shared Worker unable to reset after an update. This can break the experience on the tabs that were updated, forcing them to work with the outdated Shared Worker.

A good workaround/fix for this situation is to concatenate the KIE Sandbox version to the generated Shared Worker, this way each version can have its own worker registered on the browser.

**Comment by jomarko** [comment]
@thiagoelg hi, this is marked as good first issue, however for me, at least some pointers to the codebase would be needed to start to work on similar task.

What file is a worker implementation? What is the env_variable that we maintain the versions currently?

**Comment by thiagoelg** [comment]
I believe most changes will have to be made in the `online-editor` webpack config file, but the Shared Worker file is this one: https://github.com/kiegroup/kie-tools/blob/main/packages/online-editor/src/workspace/worker/sharedWorker.ts

Right now the current version of our packages is defined via `buildEnv.env.root.version`, available in the webpack config file if needed, and by default should be `0.0.0`, following the root package.json version.

**Comment by jomarko** [comment]
@thiagoelg thank you. I think if we speak about webpack config, we speak about this https://github.com/kiegroup/kie-tools/blob/main/packages/online-editor/webpack.config.ts#L56-L755 right?

however I am not sure what lines should be changed there , furthermore, form sharedWorker source, it si not clear to me the connection uri.

I expected we will change somewhere in codebase (in pseudocode) from `worker.connect("path/to/webworker")` to `worker.connect("path/to/webworker-specific-version")`

**Comment by thiagoelg** [comment]
Talking with @tiagobento I've learned a little bit more about shared workers, and this is what we figured out:

Open a KIE Sandbox tab and check `chrome://inspect/#workers` on your browser. There you'll see the shared worker's name (should be something like `workspaces-shared-worker`);
This is the name that needs to be changed for every release of the KIE Sandbox, and it's defined here: [workspaces-git-fs/src/worker/WorkspacesSharedWorker.ts#L56](https://github.com/kiegroup/kie-tools/blob/main/packages/workspaces-git-fs/src/worker/WorkspacesSharedWorker.ts#L56) from here [workspaces-git-fs/src/context/WorkspacesContextProvider.tsx#L43](https://github.com/kiegroup/kie-tools/blob/main/packages/workspaces-git-fs/src/context/WorkspacesContextProvider.tsx#L43).

Somehow we need to pass the KIE Sandbox version to the `WorkspacesContextProvider` so that it can concatenate it to the worker name.

So, the webpack config probably won't need to be changed.

Besides that, we need to ensure that the Serverless Logic Web Tools continues to work after this change.
